src/PekatVisionSDK/instance.py

```diff
@@ -0,0 +1,427 @@
+"""Python module for communication with PEKAT VISION 3.10.2 and higher.
+
+Author: developers@pekatvision.com
+Date:   18 May 2023
+Web:    https://github.com/pekat-vision
+"""
+
+import atexit
+import base64
+import json
+import os
+import platform
+import secrets
+import socket
+import string
+import subprocess
+import sys
+from pathlib import Path
+from typing import Literal, Optional, Union, get_args
+
+import numpy as np
+import requests
+from numpy.typing import NDArray
+from packaging import version
+
+from .errors import (
+    DistNotExistsError,
+    DistNotFoundError,
+    InvalidDataTypeError,
+    InvalidResponseTypeError,
+    NoConnectionError,
+    PekatNotStartedError,
+    PortIsAllocatedError,
+    ProjectNotFoundError,
+)
+from .result import Result
+
+StrOrPathLike = Union[str, os.PathLike]
+ResponseType = Literal["context", "image", "annotated_image", "heatmap"]
+UrlEndpoint = Literal["analyze_image", "analyze_raw_image"]
+
+ALLOWED_RESPONSE_TYPES = get_args(ResponseType)
+
+
+class Instance:
+    """A class for starting or connecting to a PEKAT VISION project."""
+
+    def __init__(
+        self,
+        project_path: Optional[StrOrPathLike] = None,
+        dist_path: Optional[StrOrPathLike] = None,
+        port: Optional[int] = None,
+        host: str = "127.0.0.1",
+        *,
+        already_running: bool = False,
+        disable_code: bool = False,
+        tutorial_only: bool = False,
+        context_in_body: bool = False,
+        wait_for_init_model: bool = False,
+        ping: bool = True,
+        gpu: int = 0,
+    ) -> None:
+        """Create an Instance object for communication with PEKAT VISION.
+
+        Arguments:
+            project_path: Path to an existing project.
+                Ignored if `already_running` is `True`.
+            dist_path: Path to PEKAT VISION installation folder.
+                Ignored if `already_running` is `True`.
+            port: Port of the project.
+            host: IP address of the project.
+                Set to `"0.0.0.0"` to listen on all interfaces.
+            already_running: Whether the project is already running.
+                If the project is not already running, project will be started.
+            disable_code: Disable code module.
+                Ignored if `already_running` is `True`.
+            tutorial_only: Only allow Anomaly module and force tutorial.
+                Ignored if `already_running` is `True`.
+            context_in_body: Whether to send the `context` `dict`  in the response body.
+                Set to `True` if the resulting context is larger than 4KB.
+            wait_for_init_model: Whether to wait for all models to be initialized before returning the [Instance][PekatVisionSDK.Instance] object.
+                Ignored if `already_running` is `False`.
```

Comment:
Isn't this ignored when `already_running` is `True` and not `False`? This seems to only be used when starting the Instance.

Reply:
Correct, gonna fix, thanks!